src/Symfony/Component/String/AbstractString.php

```diff
@@ -525,6 +525,14 @@ public function toCodePointString(): CodePointString
         return new CodePointString($this->string);
     }
 
+    /**
+     * @return non-empty-string|null
+     */
+    public function toNullOrNonEmptyString(): ?string
```

Comment:
toNonEmptyStringOrNull?

Reply:
* pro `toNonEmptyStringOrNull`: it fits better to symfony code style, where `null` always comes last in union types
* pro `toNullOrNonEmptyString`: IMO it is easier to recognize that it is an `or` condition between `null` and something else.

So ±0 from me. I will wait for a comment from the sf core team.

Reply:
I wouldn't add the `to...()` here because String component usually names methods as orders like `lower()` (instead of `toLower()`) and `reverse()` (instead of `toReverse()`).

I like Nicolas proposal (`->emptyToNull()`) and here's another one for your consideration `->nullIfEmpty()`

Reply:
> I wouldn't add the `to...()` here because String component usually names methods as orders like `lower()` (instead of `toLower()`) and `reverse()` (instead of `toReverse()`).

But all these methods return a cloned instance of the same class.
While methods with `to` prefix convert them to native types (`toString`) or other classes (`toByteString` etc.).
For me `toNonEmptyStringOrNull` is an alternative method to the existing `toString` method (return type `?string` vs `string`).

Reply:
Personally I find this name very difficult to understand --> `toNonEmptyStringOrNull()`

This method does this: "if it's an empty string, it converts it to null"

But with that method name, it looks like it does this: "convert it to a non-empty string or a null value" OK, but how does it decide which one to convert to? null or non-empty, based on what?

That's why I think `nullIfEmpty()` or `emptyToNull()` explain the purpose more clearly.

Reply:
you're overcomplicating it, you'd get a non-empty-string or null otherwise ;)

Reply:
> But all these methods return a cloned instance of the same class.

We could do the same here. 🤔 

```php
$myNullableString = $s->nullIfEmpty()?->toString();
```

Reply:
For static analysis tools it would be better to have one method with return type `non-empty-string|null`